processors/base/writers/location.py

```diff
@@ -34,7 +40,7 @@ def write_location(conn, location, source_id, trial_id=None):
     timestamp = datetime.datetime.utcnow()
 
     # Get name
-    name = helpers.clean_string(location['name'])
+    name = helpers.clean_string(extract_CanonicalName(location['name']))
```

Comment:
Instead of cleaning the output of `extract_CanonicalName`, it might be better to clean the input (i.e. `location['name']`). If we can find a match for the location, the result will already be cleaned (as it's the canonical name anyway). However, if we don't get a match, we'll return `location['name']` itself, which might need to be cleaned.

You can even do the cleaning inside the method itself, if that makes sense to you.